docs/SocialLogin.md

```diff
@@ -0,0 +1,368 @@
+        User ->> Social Server: user logs to the social server
+        Social Server ->> Zot Server: social server calls http://zot.server.domain:port/auth/gh/callback
+        Note left of Social Server: redirect to callback url, attaching state and code as query params(?code=1234567890&state=STATE)
+        Zot Server ->> Social Server: REQ - https://social-provider.com/login/oauth/token (gets token using code from prev request)
+        Social Server ->> Zot Server: RESP - returns access_token, refresh_token, period of validity using previously returned code as a query param
+        Note right of Zot Server: returns token, using previously returned code as a query param
+        Note right of Zot Server: we know token source since Zot Server made the call for it
+        Zot Server ->> Zot Server : create own JWT token to be sent to Client app with permission information
+        alt implement storing the token
+        Zot Server ->> Zot Server : stores own JWT token in memory and map it to user&auth_token of origin
+        else do not imeplement storing the token include user data
+        Zot Server ->> Zot Server : in this case the JWT token also needs to include all required user data to be used later
+        else do not imeplement storing the token include original token
+        Zot Server ->> Zot Server : in this case the JWT token also needs to include original token in order to identify the user later
+        end
+        Zot Server ->> Client app: redirects to Client App and sends the Zot Server generated JWT token
+        Client app ->> Zot Server: REQ - Ask for resource, using the Zot Server generated token for every call
+        Note right of Client app: or using PAT token for clients like Skopeo
+        alt if token storage is implemented
+        Zot Server ->> Zot Server: check user information according to memory map
+        else no token storage is implemented and zot token contains user data
+        Zot Server ->> Zot Server: decrypt Zot server generated token and extract user specific data
+        else no token storage is implemented and zot only contains original token
+        Zot Server ->> Zot Server: decrypt Zot server generated token and extract underlying auth_token, obrain information from auth providers
+        Zot Server ->> Social Server : verifies token validity and origin by calling every social provider API (if available)
+        end
+        Note left of Social Server: this verification will also be done for clients like Skopeo, which use PAT instead of Zot tokens
+        alt token invalid
+            Zot Server ->> Client app: REQ - status code 401 Unauthorized
+        else token valid
+            Zot Server ->> Client app: give resources specific to user
+        end
+```
+
+### 4: Backend with session cookies
+
+#### Short overview of the workflow
+
+    - Similar to flow#3
+    - Instead of generating JWT tokens use sessions to store information
+
```

Comment:
I think it is worth mentioning one of the great advantages of this workflow - in case of any Social provider that implements  Oauth login (so Github, Gitlab, Google) it doesn't need to verify **token validity and origin** by calling every social provider API. That is because:

1.  we query for the token on Zot server side -> we know the origin of the token by knowing what Social provider was queried
2. we get the **period of validity** as a response on the https://social-provider.com/login/oauth/token call

In case of clients that don't implement Oauth but use PATs (eg Skopeo), we have the 5th workflow.